src/Model/Serializer/ValueObject/Taxonomy/Category/CategoryDenormalizer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Model\Serializer\ValueObject\Taxonomy\Category;
+
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Category\Category;
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Category\CategoryDomain;
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Category\CategoryID;
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Category\CategoryLabel;
+use Symfony\Component\Serializer\Exception\UnsupportedException;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class CategoryDenormalizer implements DenormalizerInterface
+{
+    private ?CategoryDomain $defaultDomain;
+
+    public function __construct(?CategoryDomain $defaultDomain = null)
+    {
+        $this->defaultDomain = $defaultDomain;
+    }
+
+    public function denormalize($data, $class, $format = null, array $context = []): Category
+    {
+        if (!$this->supportsDenormalization($data, $class, $format)) {
+            throw new UnsupportedException("CategoryDenormalizer does not support {$class}.");
+        }
+
+        if (!is_array($data)) {
+            throw new UnsupportedException('Category data should be an array.');
+        }
+
+        $id = new CategoryID($data['id']);
+        $label = isset($data['label']) ? new CategoryLabel($data['label']) : null;
+        $domain = $this->createCategoryDomain($data);
+
+        return new Category($id, $label, $domain);
+    }
+
+    public function supportsDenormalization($data, $type, $format = null): bool
+    {
+        return $type === Category::class;
+    }
+
+    private function createCategoryDomain(array $data): ?CategoryDomain
+    {
+        if ($this->defaultDomain) {
+            return $this->defaultDomain;
+        }
+
+        return isset($data['domain']) ? new CategoryDomain($data['domain']) : null;
```

Comment:
I find it strange a method called "createCategoryDomain" can return NULL.
I think it should throw an exception when it cannot create a category domain.

I digged a little bit deeper and I see that this object normally goes into the constructor of `CultuurNet\UDB3\Model\ValueObject\Taxonomy\Category`. It does expect a NULL there.
But if you look further, none of the `getDomain()` functions actually check for a NULL, leading to possible NULL errors.

One random example:
```
    private function setTerms(Resource $resource, Categories $terms): void
    {
        /** @var Category $term */
        foreach ($terms as $term) {
            if ($term->getDomain()->sameAs(new CategoryDomain('eventtype'))) {
                $terms = $this->termsIriGenerator->iri($term->getId()->toString());
                $resource->set(self::PROPERTY_ACTIVITEIT_TYPE, new Resource($terms));
            }

            if ($term->getDomain()->sameAs(new CategoryDomain('theme'))) {
                $terms = $this->termsIriGenerator->iri($term->getId()->toString());
                $resource->set(self::PROPERTY_ACTIVITEIT_THEMA, new Resource($terms));
            }
        }
    }
```

Not saying you introduced this bug btw, it is already there but might be a good moment to look critical at it. :-)

Reply:
Good catch! But I'm hesitant to change it now. Maybe we need a follow up ticket for this?

Reply:
I can live with that: https://jira.publiq.be/browse/III-6433

Reply:
Great! Thanks for creating that ticket.